dashboard/src/actions/overviewActions.js

```diff
@@ -115,22 +121,31 @@ export const updateDataset =
       dispatch({ type: TYPES.LOADING });
 
       const runs = getState().overview.datasets;
-
-      const method = metaDataActions[actionType];
-
       const endpoints = getState().apiEndpoint.endpoints;
+      let method;
+      const isMultipleUpdate = Array.isArray(actionType);
+      if (!isMultipleUpdate) {
+        method = metaDataActions[actionType];
+      }
+
+      const payload = isMultipleUpdate
+        ? actionType.reduce((r, c) => Object.assign(r, c), {})
+        : { [method]: actionValue };
       const response = await API.put(
         `${endpoints?.api?.datasets_metadata}/${dataset.resource_id}`,
         {
-          metadata: { [method]: actionValue },
+          metadata: payload,
         }
       );
       if (response.status === 200) {
         const dataIndex = runs.findIndex(
           (item) => item.resource_id === dataset.resource_id
         );
-        runs[dataIndex].metadata[metaDataActions[actionType]] =
-          response.data[metaDataActions[actionType]];
+        if (!isMultipleUpdate) {
+          runs[dataIndex].metadata[metaDataActions[actionType]] =
+            response.data[metaDataActions[actionType]];
+        }
```

Comment:
Doesn't this `if` statement need an `else` which handles the multiple-update case?  (Alternatively, wouldn't it be better to iterate over `metaDataActions` and just do it for each of them -- that should work fine regardless of whether there is only one or if there are more.)

Reply:
Updated :slightly_smiling_face: